src/App.tsx

```diff
@@ -1,34 +1,158 @@
+/* eslint-disable prettier/prettier */
+/* eslint-disable @typescript-eslint/indent */
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filterSearchValue, setFilterSearchValue] = useState('');
+  const [filterStatusValue, setFilterStatusValue] = useState('all');
+
+  async function loadTodos() {
+    try {
+      const allTodos = await getTodos();
+
+      setTodos(allTodos);
+      setFilteredTodos(allTodos);
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error('Failed to load todos', error);
+    }
+
+    setIsLoading(false);
+  }
+
+  function handleOpenModal(userId: number) {
+    setSelectedUserId(userId);
+    setIsModalOpen(!isModalOpen);
```

Comment:
Toggling the modal state with `!isModalOpen` might lead to unexpected behavior if `handleOpenModal` is called multiple times. Consider setting `setIsModalOpen(true)` explicitly to ensure the modal opens correctly.